### ISSUE TITLE
No result from Web-Source based Tiles (Android)

### ISSUE THREAD

**Issue by Sebastian1989101** [pauldendulk/Mapsui#295]
Finally I got a bug / problem with Android. :D 

When I use web source based tiles (no matter from which server as long as it is a custom server) on Android, I got empty results. I tested servers from others beside my own too. It works on iOS but Android just shows a transparent map. Because of that most possible tile sources I found had an inverted Y-Axis I use this for my TileRequests: 

```
    public class TileRequest : IRequest
    {
        private readonly string _urlFormatter;

        public TileRequest(string urlFormatter)
        {
            _urlFormatter = urlFormatter;
        }

        public Uri GetUri(TileInfo info)
        {
            var y = Math.Pow(2, Convert.ToInt32(info.Index.Level)) - info.Index.Row - 1;

            var stringBuilder = new StringBuilder(_urlFormatter);
            stringBuilder.Replace("{x}", info.Index.Col.ToString(CultureInfo.InvariantCulture));
            stringBuilder.Replace("{y}", y.ToString(CultureInfo.InvariantCulture));
            stringBuilder.Replace("{z}", info.Index.Level);

            return new Uri(stringBuilder.ToString());
        }
    }
```

And this is used by my code like this: 

`return new TileLayer(new HttpTileSource(new GlobalMercator("png", 0, 6), new TileRequest(tileSourceBase + "/{z}/{x}/{y}.png"), "My Map"));`

**Comment by pauldendulk** [comment]
You have to get the tile schema definition right. The most important task of BruTile is to define that schema so the correct tiles can be fetched. https://github.com/BruTile/BruTile

By far the most common problem is the inverted Y-Axis. In BruTile you set the YAxis in this schema like this:

```
 return new HttpTileSource(new GlobalSphericalMercator(0, 18) {YAxis = YAxis.TMS},
                        "http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
                        new[] {"a", "b", "c"});
            
```

**Comment by Sebastian1989101** [comment]
@pauldendulk But the schema must be correct if it works on iOS, right? And thats my current problem. It works on iOS but not on Android (UWP not tested yet). I also see in the debugger that my TileRequest is called and that it returns the correct URI for the requested tile.

**Comment by pauldendulk** [comment]
ohw, that is ... impossible. 

The only difference between iOS and Android is in the MapControl. All the rest is in a shared library. I expect no difference in the way the tiles are fetched. 

But if there is ... perhaps some kind of images no decompressed properly on Android. I will have to look into it. Do you have a public service that does not work for you on Android?

**Comment by Sebastian1989101** [comment]
You can use the tiles from here for example: https://secure.kruse-familie.eu/botw-map/ but it is only offering Z-Index 0 to 5 and again with an inverted Y-Axis. But the problem is that Android is just displaying nothing so the display order would be irrelevant for a test.

**Comment by pauldendulk** [comment]
For me it just works.
- I set the url you post above in the OsmSample 
- Replace InfoSample.CreateMap() with OsmSample.CreateMap() in Mapsui.SampleAndroid\Activity1
- Run Mapsui.Samples.Android

Some suggestions:
- http versus https?
- You need to set some android permissions?
- Some proxy or firewall settings? Is it the same from home as from work?

**Comment by Sebastian1989101** [comment]
Maybe a permission problem. I will check it later. Which permissions are set on your test project? 

For me it happens on Simulators on my Windows and macOS device and also on a real Android phone (tested WLAN, Theatering from my iPhone and own mobile Network so I guess this is not a firewall problem).

**Comment by Sebastian1989101** [comment]
About http vs. https - This Site is only enabled for https. So it won’t work on iOS without the SSL part too.

**Comment by pauldendulk** [comment]
some more guessing. It runs on you local network which cause dns confusion somehow?

**Comment by Sebastian1989101** [comment]
Nope. It’s a real Server hosted externally. I guess I can exclude network problems. Otherwise it makes no sense that iOS works and everything else (network based) works on Android as well (even loading the point database from the exact same server and dns record).

**Comment by Sebastian1989101** [comment]
Just tested also a tile source from some other guy in the web and got the same problem with this tiles. https://tiles2.pubgmap.net/maptiles/miramar/v2/{z}/{x}/{y}.png 

For permissions, my App has the following which should include any networking: 
```
	<uses-permission android:name="com.android.vending.BILLING" />
	<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
	<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
	<uses-permission android:name="android.permission.INTERNET" />
	<uses-permission android:name="android.permission.WAKE_LOCK" />
```

**Comment by pauldendulk** [comment]
Could you try to run the sample project with your url as I described in the comment [above](https://github.com/pauldendulk/Mapsui/issues/295#issuecomment-361380258).

**Comment by Sebastian1989101** [comment]
Just tested with the Android Sample project. While the source on my server works, the one on the other don't. So know I have to check what is different and might be the problem for my new project. Maybe the other server uses some identical mechanisms. But strange it that it works on iOS.
